Extend/Controller/ThankYouController.php

```diff
@@ -0,0 +1,73 @@
+<?php
+/**
+ * Shop System Plugins:
+ * - Terms of Use can be found under:
+ * https://github.com/wirecard/oxid-ee/blob/master/_TERMS_OF_USE
+ * - License can be found under:
+ * https://github.com/wirecard/oxid-ee/blob/master/LICENSE
+ */
+
+namespace Wirecard\Oxid\Extend\Controller;
+
+use OxidEsales\Eshop\Core\Field;
+use \OxidEsales\Eshop\Core\Registry;
+
+use \Wirecard\Oxid\Core\Helper;
+use Wirecard\Oxid\Model\Transaction;
+
+/**
+ * Class ThankYouController
+ *
+ * @mixin \OxidEsales\Eshop\Application\Controller\ThankYouController
+ */
+class ThankYouController extends ThankYouController_parent
+{
+    /**
+     * Extends the parent init method
+     * deletes a wdtoken and updates the order number in the transaction table
+     */
+    public function init()
+    {
+        Registry::getSession()->deleteVariable("wdtoken");
+
+        parent::init();
+
+        $sTransactionId = $this->getOrder()->oxorder__wdoxidee_transactionid->value;
+        if ($sTransactionId) {
+            $oTransaction = oxNew(Transaction::class);
+            if ($oTransaction->loadWithTransactionId($sTransactionId)) {
+                $oTransaction->wdoxidee_ordertransactions__ordernumber
+                    = new Field($this->getOrder()->oxorder__oxordernr->value);
+                $oTransaction->save();
+            }
+        }
+    }
+
+    /**
+     * Returns if the payment is pending or not
+     *
+     * @return bool
+     */
+    public function isPaymentPending()
+    {
+        $sTransactionId = $this->getOrder()->oxorder__wdoxidee_transactionid->value;
+        if ($sTransactionId) {
+            $oTransaction = oxNew(Transaction::class);
+            if ($oTransaction->loadWithTransactionId($sTransactionId)) {
+                return strpos($oTransaction->wdoxidee_ordertransactions__type->value, 'pending') !== false;
+            }
+        }
+
+        return true;
+    }
+
+    /**
+     * Returns if the used payment method is one of the module's payment methods
+     *
+     * @return bool
+     */
+    public function isModulePaymentMethod()
```

Comment:
Don't we have this check already accessible in View_Config?

Reply:
It will be removed from there, but I will use it from order